### PR TITLE
Update inductor jobs to use CUDA 12.4

### PR DESCRIPTION
**Pull request by huydhn**
CUDA 12.4 is the default now.  This frees up some resources.  This also fixes newly added Python 3.13 job by #140733. That PR missed adding the new Docker image `pytorch-linux-focal-cuda12.4-cudnn9-py3.13-gcc9-inductor-benchmarks` into docker build workflow.

### CODE REVIEW

File: .github/workflows/inductor-unittest.yml [hunk 1]
```diff
@@ -153,3 +153,28 @@ jobs:
       docker-image: ${{ needs.linux-jammy-cpu-py3_9-gcc11-inductor-build.outputs.docker-image }}
       test-matrix: ${{ needs.linux-jammy-cpu-py3_9-gcc11-inductor-build.outputs.test-matrix }}
     secrets: inherit
+
+  linux-focal-cuda12_4-py3_13-gcc9-inductor-build:
```
Comment (huydhn) [inline .github/workflows/inductor-unittest.yml:157]: This is newly added job by https://github.com/pytorch/pytorch/pull/140733.  However, that PR missed adding the new Docker image `pytorch-linux-focal-cuda12.4-cudnn9-py3.13-gcc9-inductor-benchmarks` into docker build workflow.

File: .github/workflows/docker-builds.yml [hunk 2]
```diff
@@ -104,7 +105,7 @@ jobs:
 
       - name: Build docker image
         id: build-docker-image
-        uses: pytorch/test-infra/.github/actions/calculate-docker-image@main
+        uses: pytorch/test-infra/.github/actions/calculate-docker-image@wait-for-docker-build
```
Comment (huydhn) [inline .github/workflows/docker-builds.yml:108]: Testing https://github.com/pytorch/test-infra/pull/6013 that waits for Docker build

File: .github/workflows/_linux-build.yml [hunk 3]
```diff
@@ -136,7 +136,7 @@ jobs:
 
       - name: Calculate docker image
         id: calculate-docker-image
-        uses: pytorch/test-infra/.github/actions/calculate-docker-image@main
+        uses: pytorch/test-infra/.github/actions/calculate-docker-image@wait-for-docker-build
```
Comment (huydhn) [inline .github/workflows/_linux-build.yml:139]: Testing https://github.com/pytorch/test-infra/pull/6013 that waits for Docker build

File: .github/workflows/inductor-unittest.yml (hunk 1 above)
Comment (huydhn) [reply .github/workflows/inductor-unittest.yml:157]: I think if I should spilt the 3.13 part here into a different PR as the new docker fails to build https://github.com/pytorch/pytorch/actions/runs/12189237352/job/34004038100

**Review by atalman — APPROVED** [review]
lgtm. Thank you!

**Pull request closed without merging**